Update `browserslist` in package.json

hannalaakso opened the pull request:
This makes `browserslist` specific to the [browsers we support](https://github.com/alphagov/govuk-frontend#browser-support). Once this has been approved, we can raise a PR to make the same change in GOV.UK Design System.

Tested `/components/all` page in (if only we had cross browser visual tests 🤓):
✅ Chrome 68, 69
✅ Firefox 61, 62 (with inverted colours)
✅ Safari 9.1, 10.1, 11.1
✅ Edge 16, 17
✅ iOS Safari 9
✅ Android Chrome 63
✅ Samsung Internet 6.4.0.5
✅ Internet Explorer 8
✅ Internet Explorer 9
✅ Internet Explorer 10
✅ Internet Explorer 11

Issue originally discussed [here](https://github.com/alphagov/govuk-design-system/commit/44fac9d2c07215bb914a7bac2448dda7c1aa2872#commitcomment-30022713).

Syntax follows [browserslist docs](https://github.com/browserslist/browserslist#full-list).

Fixes https://github.com/alphagov/govuk-frontend/issues/952

https://trello.com/c/dvfqTy9D/1360-update-browser-list-in-packagejson

igloosi commented:
`-webkit` prefixes have been removed. is that expected?
<img width="857" alt="screen shot 2018-09-19 at 08 45 35" src="https://user-images.githubusercontent.com/3758555/45738472-6ff60100-bbe8-11e8-947c-b9ecc1de8df6.png">
<img width="830" alt="screen shot 2018-09-19 at 08 45 23" src="https://user-images.githubusercontent.com/3758555/45738475-71272e00-bbe8-11e8-91a2-48dd0f08fbaa.png">

hannalaakso commented:
@igloosi This is inline with the specified `browserslist`, those prefixes cater for Safari 6 which we don't officially support.

hannalaakso commented:
It sounds like we need a slightly more holistic approach: we might not want to remove prefixes for certain legacy browsers even if our browser matrix doesn't officially include them.

I've added `>0.10%` to `browserslist` which means that we add prefixes for [these browsers](http://browserl.ist/?q=%3E0.10%25) (includes Opera and UC which should be okay), in addition to our [core ones](https://github.com/alphagov/govuk-frontend#browser-support).

I've also changed support for desktop Safari to go down to 6 as we know older Safaris can be buggy and some extra ❤️ for them wouldn't hurt (moved iOS down to 8 as well). 

Tested again in:
✅ Chrome 68, 69
✅ Firefox 61, 62 (with inverted colours)
✅ Safari 6, 7.1, 9.1, 10.1, 11.1
✅ Edge 16, 17
✅ iOS Safari 9
✅ Android Chrome 63
✅ Samsung Internet 6.4.0.5
✅ Internet Explorer 8
✅ Internet Explorer 9
✅ Internet Explorer 10
✅ Internet Explorer 11

nickcolley commented on package.json:
```diff
@@ -65,10 +65,14 @@
-    "ie 8",
-    "ie 9",
-    "iOS 9"
+    ">0.10%",
```
I think I agree only provided that the extra CSS does not have a reasonable fallback.

I've not seen good evidence that by removing these attributes these browsers would receive a broken experience.

For example with flexbox we're talking, 0.01% Safari requests that would get `display: inline-block`

Your example seems hypothetical, since newer versions of Chrome do not even have prefixing as prefixing has been discontinued.

That said, don't think this is worth blocking anything over, so 👍 to whatever you choose to do.

36degrees replied:
All good stuff, I'm really just trying to think out loud here more than anything – trying to make sure that we're not being overly selective.

TBH, I wasn't aware that vendor-prefixing had 'gone away' as much as you (and [MDN](https://developer.mozilla.org/en-US/docs/Glossary/Vendor_Prefix)) suggest, so maybe this is a bit academic…

hannalaakso commented:
If we decide to keep prefixes for 0,1% and Safari and iOS  >= 9, the prefixes removed are from our current code will be:
- `-webkit-box-sizing` - not required [since Safari 5.1](https://caniuse.com/#search=box-sizing)
- `-webkit-box-shadow` - not required [since Safari 5.1](https://caniuse.com/#search=box-shadow)

hannalaakso commented:
I did a bit of investigation and for reference adding the results here. Going with `0,5%` support would affect the footer in older versions on desktop Safari as some of the `box` and `flex` prefixes for  `-webkit` would get dropped.

Safari 8 with `0,5%` support (with dropped prefixes)
<img width="1067" alt="safari 8 with 95 support" src="https://user-images.githubusercontent.com/5007934/45876663-27c71200-bd93-11e8-90c1-f580bca8ca47.png">
Safari 8 with `0,1%` support (this is what we currently have)
<img width="1107" alt="safari 8 with 99 support" src="https://user-images.githubusercontent.com/5007934/45876662-272e7b80-bd93-11e8-9806-3b439f222294.png">
So agree with @nickcolley, `0,1%` seems like a good balance for now and we can always tweak it later.
